### PR TITLE
View query

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

**Comment by tbosch** [comment]
Fix perf regression...

File: modules/@angular/core/src/view/provider.ts [hunk 1]
```diff
@@ -265,6 +312,133 @@ function checkAndUpdateProp(
...
+          break;
+        case QueryValueType.ViewContainerRef:
+          value = view.services.createViewContainerRef(view.nodes[i]);
+          break;
+        case QueryValueType.Provider:
+          value = view.nodes[i].provider.instance;
+          break;
+      }
+      values.push(value);
+    } else if (
```
Comment (tbosch) [inline modules/@angular/core/src/view/provider.ts]: remove `else`

File: modules/@angular/core/src/view/provider.ts [hunk 2]
```diff
@@ -265,6 +312,133 @@ function checkAndUpdateProp(
...
+        }
+      }
+      const projectedViews = nodeData.elementOrText.projectedViews;
+      if (projectedViews) {
+        for (let k = 0; k < projectedViews.length; k++) {
+          const projectedView = projectedViews[k];
+          calcQueryValues(projectedView, 0, projectedView.nodes.length - 1, queryId, values);
+        }
+      }
+    } else if (!(queryId in nodeDef.childMatchedQueries)) {
```
Comment (tbosch) [inline modules/@angular/core/src/view/provider.ts]: remove `else`

File: modules/@angular/core/src/view/view.ts [hunk 3]
```diff
@@ -243,7 +272,9 @@ function initView(view: ViewData, renderHost: any, component: any, context: any)
 export function checkNoChangesView(view: ViewData) {
   view.def.update(CheckNoChanges, view);
   execEmbeddedViewsAction(view, ViewAction.CheckNoChanges);
+  updateContentQueries(view);
```
Comment (tbosch) [inline modules/@angular/core/src/view/view.ts]: That's wrong! We should only check whether they changed, but not update the providers!

File: modules/@angular/core/test/view/query_spec.ts [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+        ]));
+
+        const qs: QueryService = view.nodes[1].provider.instance;
+        expect(qs.a).toBeUndefined();
+
+        checkAndUpdateView(view);
+
+        const as = qs.a.toArray();
+        expect(as.length).toBe(2);
+        expect(as[0] instanceof AService).toBe(true);
```
Comment (tbosch) [inline modules/@angular/core/test/view/query_spec.ts]: Assert which instance it is via `view.nodes`

File: modules/@angular/core/test/view/query_spec.ts [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+            elementDef(NodeFlags.None, null, 1, 'span'),
+            aServiceProvider(),
+          ])),
+        ]));
+
+        checkAndUpdateView(view);
+
+        const comp: QueryService = view.nodes[1].provider.instance;
+        expect(comp.a.length).toBe(1);
+        expect(comp.a.first instanceof AService).toBe(true);
```
Comment (tbosch) [inline modules/@angular/core/test/view/query_spec.ts]: Check instance via view.nodes

File: modules/@angular/core/src/view/provider.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/src/view/provider.ts]: Done

File: modules/@angular/core/src/view/provider.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/core/src/view/provider.ts]: Done

File: modules/@angular/core/test/view/query_spec.ts (hunk 4 above)
Comment (tbosch) [inline modules/@angular/core/test/view/query_spec.ts]: Done

File: modules/@angular/core/test/view/query_spec.ts (hunk 5 above)
Comment (tbosch) [inline modules/@angular/core/test/view/query_spec.ts]: Done

File: modules/@angular/core/src/view/view.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/core/src/view/view.ts]: Done

**Review by vicb — APPROVED** [review]

**Pull request closed without merging**